Bug : Can't open the app after setting up proxy

### Checklist

- [x] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues?q=is%3Aissue+is%3Aclosed) issues to see if someone else has already submitted the same bug report.
- [x] I will describe the problem with as much detail as possible.
- [x] If the bug only occurs with a certain link, post, image..., I will include the URL.

### App version

8.2.5

### Where did you get the app from

Other

### Android version

Android 17 on Graphene OS (I'm up to date)

### Device model

Pixel 10 Pro XL

### First occurred

_No response_

### Steps to reproduce

I set up a proxy in the app to use a public instance of Redlib and enabled it. I think I configured it right, but not so sure anymore since I can't launch the app at all (crash at startup).
From the top of my head (might be wrong) :
Method: http
Hostname: red.artemislena.eu
Port: 443

### Example post, link, markdown...

_No response_

### Expected behaviour

The app should launch and allow me to change the setting if it's not resolving the server.

### Current behaviour

The app doesn't launch at all. It just crashes when launch. I tried updating to the latest beta version but it does not work better.

### Logs

```
type: crash
osVersion: google/mustang/mustang:17/CP2A.260805.005/2026080501:user/release-keys
flags: dev options enabled
package: ml.docilealligator.infinityforreddit.plus:227, targetSdk 36
process: ml.docilealligator.infinityforreddit.plus
processUptime: 130 + 344 ms
installer: dev.imranr.obtainium

java.lang.RuntimeException: Unable to start activity ComponentInfo{ml.docilealligator.infinityforreddit.plus/ml.docilealligator.infinityforreddit.activities.MainActivity}
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4594)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4800)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:224)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:133)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:103)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:80)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:3042)
	at android.os.Handler.dispatchMessageImpl(Handler.java:142)
	at android.os.Handler.dispatchMessage(Handler.java:125)
	at android.os.Looper.loopOnce(Looper.java:296)
	at android.os.Looper.loop(Looper.java:397)
	at android.app.ActivityThread.main(ActivityThread.java:9569)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:575)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:975)
Caused by: android.os.NetworkOnMainThreadException
	at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1849)
	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:115)
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:103)
	at java.net.InetAddress.getByName(InetAddress.java:1106)
	at java.net.InetSocketAddress.<init>(InetSocketAddress.java:245)
	at ln.u1.d(SourceFile:78)
	at ln.w1.c(SourceFile:1)
	at ln.w1.b(SourceFile:9)
	at ln.w1.get(SourceFile:1)
	at hl.a.get(SourceFile:14)
	at ln.x1.b(SourceFile:3)
	at ln.x1.get(SourceFile:1)
	at hl.a.get(SourceFile:14)
	at ml.docilealligator.infinityforreddit.b$a.w3(SourceFile:3)
	at ml.docilealligator.infinityforreddit.b$a.g2(SourceFile:1)
	at ml.docilealligator.infinityforreddit.b$a.I(SourceFile:1)
	at ml.docilealligator.infinityforreddit.activities.MainActivity.onCreate(SourceFile:14)
	at android.app.Activity.performCreate(Activity.java:9393)
	at android.app.Activity.performCreate(Activity.java:9371)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1555)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4578)
	... 14 more
```

--- a/app/src/main/java/ml/docilealligator/infinityforreddit/NetworkModule.java
+++ b/app/src/main/java/ml/docilealligator/infinityforreddit/NetworkModule.java
@@ -65,7 +65,7 @@ abstract class NetworkModule {
                 String proxyHost = mProxySharedPreferences.getString(SharedPreferencesUtils.PROXY_HOSTNAME, "127.0.0.1");
                 int proxyPort = Integer.parseInt(mProxySharedPreferences.getString(SharedPreferencesUtils.PROXY_PORT, "1080"));
 
-                InetSocketAddress proxyAddr = new InetSocketAddress(proxyHost, proxyPort);
+                InetSocketAddress proxyAddr = InetSocketAddress.createUnresolved(proxyHost, proxyPort);
                 Proxy proxy = new Proxy(proxyType, proxyAddr);
                 builder.proxy(proxy);
             }

--- a/app/src/main/java/ml/docilealligator/infinityforreddit/ProxyEnabledGlideModule.java
+++ b/app/src/main/java/ml/docilealligator/infinityforreddit/ProxyEnabledGlideModule.java
@@ -36,7 +36,7 @@ public class ProxyEnabledGlideModule extends AppGlideModule {
                 String proxyHost = mProxySharedPreferences.getString(SharedPreferencesUtils.PROXY_HOSTNAME, "127.0.0.1");
                 int proxyPort = Integer.parseInt(mProxySharedPreferences.getString(SharedPreferencesUtils.PROXY_PORT, "1080"));
 
-                InetSocketAddress proxyAddr = new InetSocketAddress(proxyHost, proxyPort);
+                InetSocketAddress proxyAddr = InetSocketAddress.createUnresolved(proxyHost, proxyPort);
                 Proxy proxy = new Proxy(proxyType, proxyAddr);
                 builder.proxy(proxy);
             }

